gcloud-java-core/src/main/java/com/google/gcloud/ServiceFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud;
+
+import java.io.Serializable;
+
+public interface ServiceFactory<
+        ServiceT extends Service, 
+        OptionsT extends ServiceOptions<?, OptionsT>>
+    extends Serializable {
```

Comment:
Could we document what a ServiceFactory actually is? (There's no javadoc for it right now.) What is the use case for serialization? Then it might be clearer whether we might ever want a ServiceFactory that is not serializable.

I *am* an [expert](https://weblogs.java.net/blog/emcmanus/archive/2007/06/disassembling_s.html) on Java serialization, though I'd much rather not be. When I was working on the [JMX API](http://docs.oracle.com/javase/8/docs/api/javax/management/package-summary.html) I had to spend a ridiculous amount of time fixing problems that had been introduced through uses of serialization that were insufficiently well thought through. So that's why I have this reflex reaction of wanting to check that it's really needed.